DOC-1848: Add `DatabaseRetry` to notary metrics table (#435)
corda/corda-docs-portal: content/en/platform/corda/4.6/enterprise/notary/notary-metrics.md

{{< table >}}

|Metric Name|Type|Description|
|:-----------------------------------|:--------------------------------|:-------------------------------------------------------------------|
| `Commit` | Timer | Measures the time taken in milliseconds to commit a single transaction and the number of transactions per second (TPS).|
| `IPS` | Meter | Measures the number of committed input states per second (IPS).|

> | `Rollback` (Percona only)| Counter | Tracks the number of database transaction rollbacks. These might occur due to transient SQL exceptions, which are mitigated by retrying, or unexpected errors that cause the notarisation to be aborted.|

Reviewer: ```suggestion
| `Rollback` (Percona only)| Counter | Tracks the number of database transaction rollbacks. These might occur due to transient SQL exceptions, which are mitigated by retrying, or unexpected errors that cause the notarization to be aborted.|
```

> | `DatabaseRetry` (JPA only)| Counter | Tracks the number of attempted retries a failed transaction makes. A transaction will retry up to ten times.|

Reviewer 2: ```suggestion
| `DatabaseRetry` (JPA only)| Counter | Tracks the number of times the notary worker has attempted to retry committing a transaction to the database following a failed transaction. This might occur due to transient SQL exceptions, attempting to spend the same states on multiple notary workers in parallel, or due to issues with the underlying database.
```

I think this is a bit more accurate. I've also removed the reference to retrying up to 10 times, as this is only a default, and isn't relevant in the context of the metric.

| `NumberOfUniqueTxHashes` | Histogram | Tracks the statistical distribution of the number of unique transactions that contributed states to each transaction. This is mainly intended for trend analysis of the number of transactions a given transaction depends on.|
| `ProcessedBatchSize` | Histogram | Measures the statistical distribution of the number of states notarised per batch. The notary groups and processes states in batches for performance reasons.|
| `BatchCommit` | Timer | Measures the time taken in milliseconds to commit a single batch and the number of batches per second.|

> | `BatchSignLatency` | Timer | This metric represents the time elapsed during a batch signature, measured in milliseconds. It includes building the merkle tree for all the participating transactions using a cryptographic hash function and then signing over the root. |

Reviewer: ```suggestion
| `BatchSignLatency` | Timer | Represents the time elapsed during a batch signature, measured in milliseconds. It includes building the Merkle tree for all the participating transactions using a cryptographic hash function and then signing over the root. |
```
